### PR TITLE
Extract wire message handling into a method.

### PR DESCRIPTION
**Pull request by arik-so**
This is a response to splitting #585 into smaller components. This extraction should allow the subsequent creation of a trait for all message handling, thereby enabling more flexibility in the state machine, particularly for bindings.

### CODE REVIEW

File: lightning/src/ln/peer_handler.rs [hunk 1]
```diff
@@ -203,6 +203,12 @@ pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, L: Deref> where
 	logger: L,
 }
 
+enum MessageHandlingResult {
+	Ok,
+	PeerHandleError(PeerHandleError),
+	LightningError(LightningError),
+}
+
```
[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/ln/peer_handler.rs:211]: You're right, that's a much better approach

**Comment by valentinewallace** [comment]
Second all of Jeff's comments but this looks like a nice change. Might appreciate a high-level comment with what the end goal `trait` looks like to make sure I understand correctly what you're going for -- and my understanding is that the PeerHandler will have a pointer to an implementer of this trait?

File: lightning/src/ln/peer_handler.rs [hunk 2]
```diff
@@ -606,177 +623,14 @@ impl<Descriptor: SocketDescriptor, CM: Deref, L: Deref> PeerManager<Descriptor,
...
-												return Err(PeerHandleError{ no_connection_possible: true });
-											},
-											wire::Message::Unknown(msg_type) => {
-												log_trace!(self.logger, "Received unknown odd message of type {}, ignoring", msg_type);
-											},
+										if let Err(handling_error) = self.handle_message(&mut peers.peers_needing_send, peer, peer_descriptor.clone(), message){
+											match handling_error {
+												MessageHandlingError::PeerHandleError(e) => { return Err(e) },
+												MessageHandlingError::LightningError(e) => {
+													// use the macro to handle the message actions
```
Comment (valentinewallace) [inline lightning/src/ln/peer_handler.rs:630]: nit: I'd get rid of this comment

File: lightning/src/ln/peer_handler.rs [hunk 3]
```diff
@@ -796,6 +650,194 @@ impl<Descriptor: SocketDescriptor, CM: Deref, L: Deref> PeerManager<Descriptor,
...
+				}
+				if msg.features.requires_unknown_bits() {
+					log_info!(self.logger, "Peer local features required unknown version bits");
+					return Err(PeerHandleError{ no_connection_possible: true }.into());
+				}
+				if peer.their_features.is_some() {
+					return Err(PeerHandleError{ no_connection_possible: false }.into());
+				}
+
+				log_info!(
```
[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/ln/peer_handler.rs:687]: I rephrased this

File: lightning/src/ln/peer_handler.rs [hunk 4]
```diff
@@ -606,194 +612,224 @@ impl<Descriptor: SocketDescriptor, CM: Deref, L: Deref> PeerManager<Descriptor,
...
-												if !peer.outbound {
-													let mut features = InitFeatures::known();
-													if !self.message_handler.route_handler.should_request_full_sync(&peer.their_node_id.unwrap()) {
-														features.clear_initial_routing_sync();
-													}
+										let new_output_buffer_size = peer.pending_outbound_buffer.len();
+										if new_output_buffer_size > output_buffer_size {
+											// we have enqueued some messages, let's make the peers aware of it
+											peers.peers_needing_send.insert(peer_descriptor.clone());
+										}
```
[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/ln/peer_handler.rs]: Yup, you're right.

File: lightning/src/ln/peer_handler.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/ln/peer_handler.rs]: (done)

**Pull request merged**